CMakeLists.txt

```diff
@@ -1,23 +1,65 @@
 cmake_minimum_required(VERSION 3.12) #Python3 package locator only added in 3.12
 
+project(cmakedoc DESCRIPTION "CMake automatic documentation generator" LANGUAGES NONE) #Not using any compiled languages so don't list, removes check for compilers
+
 find_package(Python3 COMPONENTS Interpreter)
 
+#[[[
+# Custom function to locate a Python module. Requires the Python interpreter to be located first.
+#
+# Sets PY_${module_upper} to the module file's location if found.
+#
+# :param module: Python module name to locate
+# :param REQUIRED: Optional paramater, specifies that this module is required for continued execution.
+# Sets ${module}_FIND_REQUIRED to true.
+#
+#]]
+function(find_python_module module)
```

Comment:
It's becoming more common in CMake to have a directory `cmake` (in the root of the repo) that contains additional CMake modules. I'd recommend putting your function in a file `cmake/find_python_module.cmake` and then doing `include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/find_python_module.cmake)` here (I always forget the include rules for `include` you may be able to get away without an absolute path and the `*.cmake` suffix)

Reply:
Fixed